### PR TITLE
Log sftp file and dir info for debugging

### PR DESCRIPTION
**Pull request by TimboTambo**
# Log sftp file and dir info for debugging

Relates to #619
Log info about all files in the request file's containing directory, along with permissions info, in order to better diagnose sftp "File could not be opened" issue.

### Acceptance Criteria

- [ ] Info about file and dir written to logs

### Testing information

- Check in file
- Observe logs to see file and dir info written

### Pre-flight Checklist

- [ ] Testing aligns with [testing strategy][testing strategy]
- [ ] DUXD review approval
- [ ] Accessibility Reviewed
- [ ] Product owner demo

### Pre-merge Checklist

- [ ] Branch test approved

[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"

### CODE REVIEW

File: medicines/doc-index-updater/src/create_manager/sftp_client.rs [hunk 1]
```diff
@@ -87,6 +87,23 @@ async fn retrieve_file_from_sftp(
         Err(_) => path,
     };
     tracing::info!("{:?}", path);
+
+    // Additional logging to debug observed sftp issue
+    // in nonprod environment
+    let parent_dir = path.parent();
+    if let Some(parnet_dir_path) = parent_dir {
```
Comment (craiga) [inline medicines/doc-index-updater/src/create_manager/sftp_client.rs:94]: Should be `parent_dir_path`?

File: medicines/doc-index-updater/src/create_manager/sftp_client.rs [hunk 2]
```diff
@@ -87,6 +87,23 @@ async fn retrieve_file_from_sftp(
...
+        tracing::debug!("Finding contents of {:?}", &parnet_dir_path);
+        if let Ok(file_stats) = sftp.readdir(parnet_dir_path) {
+            for (path_buf, file_stat) in file_stats {
+                tracing::debug!("{:?}", path_buf.to_str());
```
Comment (craiga) [inline medicines/doc-index-updater/src/create_manager/sftp_client.rs:98]: Is this outputting the file name or file contents?

File: medicines/doc-index-updater/src/create_manager/sftp_client.rs [hunk 3]
```diff
@@ -87,6 +87,23 @@ async fn retrieve_file_from_sftp(
         Err(_) => path,
     };
     tracing::info!("{:?}", path);
+
+    // Additional logging to debug observed sftp issue
```
Comment (majikandy) [inline medicines/doc-index-updater/src/create_manager/sftp_client.rs:91]: Do we avoid comments where at all possible and make the code alone communicate?

Comment (TimboTambo) [reply medicines/doc-index-updater/src/create_manager/sftp_client.rs:91]: I just meant this to indicate that it's temporary, linked to debuggin this issue, so should be removed in clean-up.

Comment (craiga) [reply medicines/doc-index-updater/src/create_manager/sftp_client.rs:91]: FWIW, I like comments.

File: medicines/doc-index-updater/src/create_manager/sftp_client.rs (hunk 2 above)
Comment (TimboTambo) [reply medicines/doc-index-updater/src/create_manager/sftp_client.rs:98]: It's the file path.

Comment (majikandy) [reply medicines/doc-index-updater/src/create_manager/sftp_client.rs:98]: and maybe expand buf -> buffer ?    if this code is only temporary then no big deal

File: medicines/doc-index-updater/src/create_manager/sftp_client.rs (hunk 3 above)
Comment (majikandy) [reply medicines/doc-index-updater/src/create_manager/sftp_client.rs:91]: cool @TimboTambo - that's fine then

File: medicines/doc-index-updater/src/create_manager/sftp_client.rs (hunk 1 above)
Comment (TimboTambo) [reply medicines/doc-index-updater/src/create_manager/sftp_client.rs]: Yep!

**Pull request merged**